include/bout/mesh.hxx

```diff
@@ -427,15 +428,43 @@ class Mesh {
   bool IncIntShear; ///< Include integrated shear (if shifting X)
 
   /// Coordinate system
-  Coordinates *coordinates() {
-    if (coords) { // True branch most common, returns immediately
-      return coords;
+  Coordinates *coordinates(const CELL_LOC location = CELL_DEFAULT) {
+    switch (location) { // True branch most common, returns immediately
+    case CELL_CENTRE: if (coords_centre) return coords_centre;
+    case CELL_XLOW: if (coords_xlow) return coords_xlow;
```

Comment:
I think this could lead to bugs -- say the user requests XLOW but it is `nullptr` we won't hit the `return coords_xlow` but instead drop through to the next case, which means if `coords_ylow` is not `nullptr`, for example, then we'll end up returning that instead without the calling code having any means to tell.

Reply:
will get rid of this by changing to use std::map